src/main/java/org/isf/config/SecurityConfig.java

```diff
@@ -108,102 +108,169 @@ public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+						// hospitals
+						.antMatchers(HttpMethod.POST, "/hospitals/**").hasAuthority("admin")
+						.antMatchers(HttpMethod.PUT, "/hospitals/**").hasAuthority("admin")
 						.antMatchers(HttpMethod.DELETE, "/hospitals/**").hasAuthority("admin")
 						.antMatchers(HttpMethod.PATCH, "/hospitals/**").hasAuthority("admin")
-						//.antMatchers(HttpMethod.GET, "/hospitals/**").hasAnyAuthority("admin", "guest")
-						.antMatchers(HttpMethod.POST, "/laboratories/**").hasAuthority("admin").antMatchers(HttpMethod.PUT, "/laboratories/**")
-						.hasAuthority("admin").antMatchers(HttpMethod.DELETE, "/laboratories/**").hasAuthority("admin")
-						.antMatchers(HttpMethod.PATCH, "/laboratories/**").hasAuthority("admin").antMatchers(HttpMethod.GET, "/laboratories/**")
-						.hasAnyAuthority("admin", "guest")
+						// .antMatchers(HttpMethod.GET, "/hospitals/**").hasAnyAuthority("admin", "guest") to anyone
+						// laboratories
+						.antMatchers(HttpMethod.POST, "/laboratories/**").hasAuthority("admin")
+						.antMatchers(HttpMethod.PUT, "/laboratories/**").hasAuthority("admin")
+						.antMatchers(HttpMethod.DELETE, "/laboratories/**").hasAuthority("admin")
+						.antMatchers(HttpMethod.PATCH, "/laboratories/**").hasAuthority("admin")
+						.antMatchers(HttpMethod.GET, "/laboratories/**").hasAnyAuthority("admin", "guest")
 						// .antMatchers("/auth-needed/**").authenticated()
 						// .antMatchers("/noauth-public/**").permitAll()
 						// .antMatchers("/admin/**").hasAuthority("admin")
 						// age types
-						.antMatchers(HttpMethod.PUT, "/agetypes/**").hasAuthority("admin").antMatchers(HttpMethod.GET, "/agetypes/**")
-						.hasAnyAuthority("admin", "guest")
+						.antMatchers(HttpMethod.PUT, "/agetypes/**").hasAuthority("admin")
+						.antMatchers(HttpMethod.GET, "/agetypes/**").hasAnyAuthority("admin", "guest")
 						// disease types
-						.antMatchers(HttpMethod.POST, "/diseasetypes/**").hasAuthority("admin").antMatchers(HttpMethod.PUT, "/diseasetypes/**")
-						.hasAuthority("admin").antMatchers(HttpMethod.DELETE, "/diseasetypes/**").hasAuthority("admin")
+						.antMatchers(HttpMethod.POST, "/diseasetypes/**").hasAuthority("admin")
+						.antMatchers(HttpMethod.PUT, "/diseasetypes/**").hasAuthority("admin")
+						.antMatchers(HttpMethod.DELETE, "/diseasetypes/**").hasAuthority("admin")
 						.antMatchers(HttpMethod.GET, "/diseasetypes/**").hasAnyAuthority("admin", "guest")
 						// opd
-						.antMatchers(HttpMethod.POST, "/opds/**").permitAll().antMatchers(HttpMethod.PUT, "/opds/**").permitAll()
-						.antMatchers(HttpMethod.DELETE, "/opds/**").permitAll().antMatchers(HttpMethod.PATCH, "/opds/**").permitAll()
+						.antMatchers(HttpMethod.POST, "/opds/**").permitAll()
+						.antMatchers(HttpMethod.PUT, "/opds/**").permitAll()
+						.antMatchers(HttpMethod.DELETE, "/opds/**").permitAll()
+						.antMatchers(HttpMethod.PATCH, "/opds/**").permitAll()
```

Comment:
Hi @ArnaudFonzam, any reason why from this point they are all `permitAll()` except `GET` methods?

Reply:
Hello @mwithi no we can also fix permitAll() for Get methods.

Reply:
what do you mean? GET got always (correctly) `hasAnyAuthority("admin", "guest")`
I mean that only others need to get `hasAnyAuthority("admin")`

After of course we should start to manage roles in the DB and in the code

Reply:
Okay for me not problem we can remove guest and do it like you suggest.

Reply:
Guest user can remain, the point is to apply these changes that should be consistent now, please review this PR. Thanks!